File: ecommerce-api-legacy/src/controllers/reportController.js
'use strict';

/** Controller do relatório financeiro. */
class ReportController {
  constructor({ reportService }) {
    this.service = reportService;
  }

  financeiro = async (_req, res) => {
    res.status(200).json(await this.service.financeiro());
  };
}

module.exports = ReportController;

File: task-manager-api/controllers/report_controller.py
"""Controller de relatórios."""
from flask import jsonify


class ReportController:
    def __init__(self, report_service):
        self.service = report_service

    def resumo(self):
        return jsonify(self.service.resumo()), 200

    def por_usuario(self, user_id):
        return jsonify(self.service.por_usuario(user_id)), 200
